client/shipping/shipping-recommendations.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, Children } from '@wordpress/element';
+import { Text } from '@woocommerce/experimental';
+import { PLUGINS_STORE_NAME } from '@woocommerce/data';
+import ExternalIcon from 'gridicons/dist/external';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore VisuallyHidden is present, it's just not typed
+// eslint-disable-next-line @woocommerce/dependency-group
+import { CardFooter, Button, VisuallyHidden } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { createNoticesFromResponse } from '../lib/notices';
+import {
+	DismissableList,
+	DismissableListHeading,
+} from '../settings-recommendations/dismissable-list';
+import WooCommerceServicesItem from './woocommerce-services-item';
+import './shipping-recommendations.scss';
+
+const useInstallPlugin = () => {
+	const [ pluginsBeingSetup, setPluginsBeingSetup ] = useState<
+		Array< string >
+	>( [] );
+
+	const { installAndActivatePlugins } = useDispatch( PLUGINS_STORE_NAME );
+
+	const handleSetup = ( slugs: string[] ): PromiseLike< void > => {
+		if ( pluginsBeingSetup.length > 0 ) {
+			return Promise.resolve();
+		}
+
+		setPluginsBeingSetup( slugs );
+
+		return installAndActivatePlugins( slugs )
+			.then( () => {
+				setPluginsBeingSetup( [] );
+			} )
+			.catch( ( response: { errors: Record< string, string > } ) => {
+				createNoticesFromResponse( response );
+				setPluginsBeingSetup( [] );
+
+				return Promise.reject();
+			} );
+	};
+
+	return [ pluginsBeingSetup, handleSetup ] as const;
+};
+
+const ShippingRecommandationsList: React.FC = ( { children } ) => (
+	<DismissableList
+		className="woocommerce-recommended-shipping-extensions"
+		dismissOptionName="woocommerce_setting_shipping_recommendations_hidden"
```

Comment:
Nitpicky but maybe `woocommerce_shipping_recommendations_settings_hidden`?  I think `shipping_recommendations_` is the prefix here unless we have precedence with `woocommerce_settings_` as a prefix.

Reply:
I was following along with what already existed here: https://github.com/woocommerce/woocommerce-admin/blob/f5e2cee3e84dc369cde2a56fcb3e026232552207/client/payments/payment-recommendations.tsx#L33

I can update this tho if you'd prefer 🤷

Reply:
Rereading this with fresh eyes, I think `woocommerce_settings_` makes sense (though I think pluralized settings may be slightly more semantic?) since these are not "shipping settings" but instead are "shipping recommendations" on the settings screen.